WeakAurasTemplates/TriggerTemplates.lua

```diff
@@ -912,30 +920,64 @@ function WeakAuras.CreateTemplateView(frame)
           button:SetIcon(item.icon);
         end
         button:SetClick(function()
-          local subTypes = subTypesFor(item, newView.data.regionType);
-          if #subTypes < 2 then
-            local subType = subTypes[1] or {}
-            if (newView.existingAura) then
-              newView.choosenItem = item;
-              newView.choosenSubType = subType;
-              createButtons();
-            else
-              replaceTrigger(newView.data, item, subType);
-              replaceCondition(newView.data, item, subType);
-              newView.data.id = WeakAuras.FindUnusedId(item.title);
-              newView.data.load = {};
-              if (item.load) then
-                WeakAuras.DeepCopy(item.load, newView.data.load);
+          if ((IsControlKeyDown() or newView.BatchMode) and not newView.existingAura) then
+            local isPicked = false;
+            for k,v in pairs(newView.choosenItemBatch) do
+              if v.spell == item.spell then
+                isPicked = true;
               end
-              newView:CancelClose();
-              WeakAuras.Add(newView.data);
-              WeakAuras.NewDisplayButton(newView.data);
-              WeakAuras.PickDisplay(newView.data.id);
+            end
+            if isPicked then
+              button.frame:UnlockHighlight();
+              for k,v in pairs(newView.choosenItemBatch) do
+                if v.spell == item.spell then
+                  newView.choosenItemBatch[k] = nil;
+                end
+              end
+              newView.choosenItemButtonsBatch[j] = nil;
+            else
+              button.frame:LockHighlight();
+              tinsert(newView.choosenItemBatch, item);
```

Comment:
If you use chooseItemBatch as a hash instead of a array, the code should get easier.

That is, use: choosenItemBatch[item] = true here;

To clear use:
choosenItemBatch[item] = nil;

And to check whether it is picked use if (choosenItemBatch[item])

Reply:
you are a genius